### PR TITLE
refactor: Improve Android 10 stability

### PR DESCRIPTION
**Pull request by eliaslecomte**
- Adds Android 10 disconnect (via WifiUtils).
- Adds more fine grained errors for ForceWifiUsage, connect, disconnect, remove.
- Detects unconnectable oneplus/honor devices and throws an error on connect of it.

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -30,12 +30,14 @@ repositories {
...
+    maven { url 'https://jitpack.io' }
 }
 
 dependencies {
     //noinspection GradleDynamicVersion
     implementation 'com.facebook.react:react-native:+'
 
-    implementation 'com.thanosfisherman.wifiutils:wifiutils:1.6.1'
+
+    implementation 'com.github.eliaslecomte:WifiUtils:ffbdc86'
```
Comment (eliaslecomte) [inline android/build.gradle:41]: I want to switch to WifiUtils as soon as they release a new version (code has been merged in master already).

Comment (Rapsssito) [reply android/build.gradle:41]: @eliaslecomte, should we wait until a new version of WifiUtils is released for merging this PR?

Comment (eliaslecomte) [reply android/build.gradle:41]: I think that's best :-). Let's see if I can get in touch.

**Comment by eliaslecomte** [comment]
I see I didn't update the readme completely. Still have to do that.

**Comment by eliaslecomte** [comment]
WifiUtils dependency and readme are updated :-).

**Pull request merged**